military_features/scripts/AppendMilitaryFeatures.py

```diff
@@ -0,0 +1,123 @@
+    debug = True # switch off if detailed info not needed
+    
+    # LICENSE ==========================================
+        
+    # Check that we have Advanced/ArcInfo
+    licenseState = arcpy.CheckProduct("ArcInfo")
+    arcpy.AddMessage("ArcGIS Advanced/ArcInfo license state = " + licenseState)
+    if (licenseState == "Unavailable") or (licenseState == "NotLicensed") :
+        arcpy.AddError("Editing Representation Rules requires Desktop Advanced License (ArcInfo) - Tool can't continue")    
+        raise Exception('License Error')
+            
+    # MAIN ============================================    
+            
+    # set command line arguments - Note: need quotes("") around command line arguments (to handle spaces)
+    arguments = "\"" + str(inputTable) + "\" " + "\"" + str(targetGeodatabase) \
+        + "\" " + str(sidcField) + " " + str(standard)
+    
+    # set command string
+    # AppendMilitaryFeatures.exe <inputFeatureClass> <outputGDB> [sicField]
+    currentPath = os.path.dirname(__file__)
+    EXE_PATH = os.path.normpath(os.path.join(currentPath, r"../application/"))
+    EXE_PATH_AND_FILE = "\"" + os.path.join(EXE_PATH, r"AppendMilitaryFeatures.exe") + "\""
+    EXE_CMD = EXE_PATH_AND_FILE + r" " + arguments
+    LOG_FILENAME = os.path.normpath(os.path.join(EXE_PATH, r"log.txt"))
+    COMMAND_LINE = EXE_CMD + r" > " + "\"" + LOG_FILENAME + "\""
+    # Ex: COMMAND_LINE=r"AppendMilitaryFeatures.exe arg1 arg2 arg3 > c:/temp/log.txt"
+    
+    # Run the command at a command prompt
+    if debug == True: arcpy.AddMessage("Command string: \n" + str(COMMAND_LINE))
+
+    # Popen Version with Return Code:
+    proc = subprocess.Popen(COMMAND_LINE, shell=True) 
+    proc.wait()
+    retCode = proc.returncode
+
+    # Send entire console log file to ArcPy Messages (so we know what happened):
+    logFile = open(LOG_FILENAME, "r") 
+    for line in logFile: 
+        nonewline = line.strip('\n')
+        arcpy.AddMessage(nonewline) 
+
+    osmsg = "Return Code: " + str(retCode)
+    arcpy.AddMessage(osmsg)
+
+    if (retCode > 0) and (retCode < 8) :
+        errorDictionary = {}
+        errorDictionary[0] = 'No Error'
+        errorDictionary[1] = 'Failed to load dependent data files'
+        errorDictionary[2] = 'Input Dataset does not exist/can not be opened'
+        errorDictionary[3] = 'No military features found in input'
+        errorDictionary[4] = 'Output GDB does not exist/can not be opened'
+        errorDictionary[5] = 'Exclusive Schema Lock could not be obtained on Output GDB'
+        errorDictionary[6] = 'No [SIDC] field in input data'
+        errorDictionary[7] = 'Could not find required Style Files - check ArcGIS Styles folder'
+
+        arcpy.AddError(errorDictionary[retCode])
+    
+    # Set output
+    arcpy.SetParameter(4, targetGeodatabase)
+
+except arcpy.ExecuteError: 
+    # Get the tool error messages 
+    msgs = arcpy.GetMessages() 
+    arcpy.AddError(msgs) 
+    print msgs
```

Comment:
2.x to 3.x: print as function instead of statement: print(msgs)

Reply:
These are 10.X/Python 2 tools with no possibility of Pro use